RadixWallet/Clients/FactorSourcesClient/FactorSourcesClient+Interface.swift

```diff
@@ -244,46 +248,48 @@ extension FactorSourcesClient {
 			label: label
 		)
 
-		_ = try await addPrivateHDFactorSource(.init(
-			factorSource: factorSource.embed(),
-			mnemonicWithPasshprase: mnemonicWithPassphrase,
-			saveIntoProfile: true
-		))
+		try await saveFactorSource(factorSource.embed())
 
 		return factorSource.embed()
 	}
 
 	@discardableResult
 	public func addOnDeviceFactorSource(
 		privateHDFactorSource: PrivateHDFactorSource,
+		onMnemonicExistsStrategy: ImportMnemonic.State.PersistStrategy.OnMnemonicExistsStrategy,
 		saveIntoProfile: Bool
 	) async throws -> FactorSourceID {
-		try await addPrivateHDFactorSource(.init(
-			factorSource: privateHDFactorSource.factorSource.embed(),
-			mnemonicWithPasshprase: privateHDFactorSource.mnemonicWithPassphrase,
-			saveIntoProfile: saveIntoProfile
-		))
+		try await addPrivateHDFactorSource(
+			.init(
+				privateHDFactorSource: privateHDFactorSource,
+				onMnemonicExistsStrategy: onMnemonicExistsStrategy,
+				saveIntoProfile: saveIntoProfile
+			)
+		).embed()
 	}
 
 	public func addOnDeviceFactorSource(
-		onDeviceMnemonicKind: MnemonicBasedFactorSourceKind.OnDeviceMnemonicKind,
+		onDeviceMnemonicKind: FactorSourceKindOfMnemonic.OnDeviceMnemonicKind,
 		mnemonicWithPassphrase: MnemonicWithPassphrase,
-		saveIntoProfile: Bool? = nil
-	) async throws -> FactorSource {
+		onMnemonicExistsStrategy: ImportMnemonic.State.PersistStrategy.OnMnemonicExistsStrategy,
+		saveIntoProfile: Bool
+	) async throws -> DeviceFactorSource {
 		let isOlympiaCompatible = onDeviceMnemonicKind == .olympia
-		let shouldSaveIntoProfile: Bool = saveIntoProfile ?? isOlympiaCompatible
 
 		let factorSource: DeviceFactorSource = try isOlympiaCompatible
 			? .olympia(mnemonicWithPassphrase: mnemonicWithPassphrase)
 			: .babylon(mnemonicWithPassphrase: mnemonicWithPassphrase)
 
-		_ = try await addPrivateHDFactorSource(.init(
-			factorSource: factorSource.embed(),
-			mnemonicWithPasshprase: mnemonicWithPassphrase,
-			saveIntoProfile: shouldSaveIntoProfile
-		))
+		_ = try await self.addOnDeviceFactorSource(
```

Comment:
do we need the `_ = ` when it's `@discardableResult`?

Reply:
fixed